workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/DialogSession.kt

```diff
@@ -46,14 +49,17 @@ internal class DialogSession(
    * Wrap the given dialog holder to maintain [allowEvents] on each update.
    */
   val holder: OverlayDialogHolder<Overlay> = OverlayDialogHolder(
-    holder.environment, holder.dialog
+    holder.environment, holder.dialog, holder.onUpdateBounds, holder.onBackPressed
   ) { overlay, environment ->
     allowEvents = !environment[CoveredByModal]
     holder.show(overlay, environment)
   }
 
   val savedStateRegistryKey = Compatible.keyFor(holder.showing, index.toString())
 
+  private val KeyEvent.isBackPress: Boolean
+    get() = (keyCode == KEYCODE_BACK || keyCode == KEYCODE_ESCAPE) && action == ACTION_UP
```

Comment:
Hmm. Seems fragile. I guess if we have to check like this then better to have it centralized.

Reply:
It's pretty well proven, we've been relying on this fragment for years now.